index.js

```diff
@@ -797,17 +797,16 @@
   //. the type comprising every [`String`][] value except `''`.
   //.
   //. The given type must satisfy the [Monoid][] and [Setoid][] specifications.
-  var NonEmpty = UnaryType(
-    'sanctuary-def/NonEmpty',
-    functionUrl('NonEmpty'),
-    Any,
-    function(x) {
+  function NonEmpty(parent) {
+    function test(x) {
       return Z.Monoid.test(x) &&
              Z.Setoid.test(x) &&
              !Z.equals(x, Z.empty(x.constructor));
-    },
-    function(monoid) { return [monoid]; }
-  );
+    }
+    function extract(monoid) { return [monoid]; }
+    var t = UnaryTypeWithUrl('sanctuary-def/NonEmpty', parent, test, extract);
```

Comment:
This is particularly problematic, because the property names described by `propPath` might exist in both the parent type and the refinement, and `validate` does not tell us which it was.